frontend/src/metabase/static-viz/lib/svg.ts

```diff
@@ -22,9 +27,42 @@ const transformSvgForOutline = (svgString: string) => {
   );
 };
 
-// Removing xmlns attributes that cause Batik failing to convert svg to png
-export const sanitizeSvgForBatik = (svg: string) => {
-  return transformSvgForOutline(svg)
+/**
+ * Fixes `<text>` elements not being vertically centered due to Batik not
+ * supporting the `dominant-baseline` property.
+ */
+export function patchDominantBaseline(svgString: string) {
+  const svgElem = fromHtml(svgString, { fragment: true, space: "svg" })
+    .children[0] as Element;
+
+  svgElem.children.forEach(child => {
```

Comment:
Should we traverse and transform even nested `text` nodes?

Reply:
Good point, I forgot that some nodes could be nested. I'll add this.

Reply:
@alxnddr I've made it recursive and added a unit test, please have a look